LibTest/js_interop/JSAnyUtilityExtension/isA_A02_t01.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion bool isA<T extends JSAny?>()
+/// Whether this [JSAny?] is an instance of the JavaScript type that is declared
+/// by `T`.
+/// ...
+/// If `T` is a primitive JS type like [JSString], this uses a `typeof` check
+/// that corresponds to that primitive type like `typeofEquals('string')`.
+///
+/// @description Checks that `isA` works as expected for JS primitive types.
+/// @author sgrekhov22@gmail.com
+
+import 'dart:js_interop';
+import 'dart:js_interop_unsafe';
+import '../../../Utils/expect.dart';
+import '../js_utils.dart';
+
+main() {
+  eval(r'''
+    globalThis.s = "JS string";
+    globalThis.b = false;
+    globalThis.n1 = 42;
+    globalThis.n2 = 3.14;
+  ''');
+  Expect.isTrue(globalContext["s"].isA<JSString>());
+  Expect.isTrue(globalContext["s"].isA<JSAny>());
+  Expect.isFalse(globalContext["s"].isA<JSObject>());
```

Comment:
Oh, so a JavaScript string is not a `JSObject`? Surprising!

The [`JSObject` documentation](https://api.dart.dev/dart-js_interop/JSObject-extension-type.html) does say that it "is the supertype of all JavaScript objects, but not other JS types, like primitives", but the [documentation of `JSString`](https://api.dart.dev/dart-js_interop/JSString-extension-type.html) does not indicate that `JSString` is primitive.

Should it? It could be considered to be implied by the fact that 'Implemented types' does not include `JSObject`, but that's still less than 100% explicit. @srujzs, WDYT?

Reply:
`JSBigInt` and `JSSymbol` are added to the matrix in this test.